dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/utils/DagHelper.java

```diff
@@ -424,19 +424,61 @@ private static List<Long> skipTaskNode4Switch(TaskNode taskNode,
         int resultConditionLocation = switchParameters.getResultConditionLocation();
         List<SwitchResultVo> conditionResultVoList = switchParameters.getDependTaskList();
         List<Long> switchTaskList = conditionResultVoList.get(resultConditionLocation).getNextNode();
+        Set<Long> switchNeedWorkCodes = new HashSet<>();
         if (CollectionUtils.isEmpty(switchTaskList)) {
             switchTaskList = new ArrayList<>();
+        } else {
+            // get all downstream nodes of the branch that the switch node needs to execute
+            for (Long switchTaskCode : switchTaskList) {
+                getSwitchNeedWorkCodes(switchTaskCode, dag, switchNeedWorkCodes);
+            }
         }
         // conditionResultVoList.remove(resultConditionLocation);
         for (SwitchResultVo info : conditionResultVoList) {
             if (CollectionUtils.isEmpty(info.getNextNode())) {
                 continue;
             }
-            setTaskNodeSkip(info.getNextNode().get(0), dag, completeTaskList, skipTaskNodeList);
+            setSwitchTaskNodeSkip(info.getNextNode().get(0), dag, completeTaskList, skipTaskNodeList,
```

Comment:
`info.getNextNode().get(0)` is skipNode? If true, I think it's better to add method like `getFirstSkipNode` for  `SwitchResultVo`, WDYT?

Reply:
i will improve this

Reply:
hi @caishunfeng  according to the front-end page, it seems that a switchResult has only one nextNode. So in order to avoid ambiguity, use a loop to traverse (instead of ``` get(0)```), what do you think?